WordPress/Classes/ViewRelated/Post/PostListViewController.m

```diff
@@ -0,0 +1,1319 @@
+            return PostAuthorFilterEveryone;
+        }
+    }
+    return PostAuthorFilterMine;
+}
+
+- (void)setCurrentPostAuthorFilter:(PostAuthorFilter)filter
+{
+    if (filter == [self currentPostAuthorFilter]) {
+        return;
+    }
+    [[NSUserDefaults standardUserDefaults] setObject:@(filter) forKey:CurrentPostAuthorFilterKey];
+    [NSUserDefaults resetStandardUserDefaults];
+    [self.recentlyTrashedPostIDs removeAllObjects];
+    [self updateAndPerformFetchRequestClearingCachedRowHeights:YES];
+}
+
+- (PostListFilter *)currentPostListFilter
+{
+    return self.postListFilters[[self currentFilterIndex]];
+}
+
+- (PostListFilter *)filterThatDisplaysPostsWithStatus:(NSString *)postStatus
+{
+    for (PostListFilter *filter in self.postListFilters) {
+        if ([filter.statuses containsObject:postStatus]) {
+            return filter;
+        }
+    }
+    // The draft filter is the catch all by convention.
+    return [self.postListFilters objectAtIndex:[self indexForFilterWithType:PostListStatusFilterDraft]];
+}
+
+- (NSInteger)indexForFilterWithType:(PostListStatusFilter)filterType
+{
+    NSInteger index = [self.postListFilters indexOfObjectPassingTest:^BOOL(id obj, NSUInteger idx, BOOL *stop) {
+        PostListFilter *filter = (PostListFilter *)obj;
+        return filter.filterType == filterType;
+    }];
+    return index;
+}
+
+- (NSInteger)currentFilterIndex
+{
+    NSNumber *filter = [[NSUserDefaults standardUserDefaults] objectForKey:CurrentPostListStatusFilterKey];
+    if (!filter || [filter integerValue] >= [self.postListFilters count]) {
+        return 0; // first item is the default
+    }
+    return [filter integerValue];
+}
+
+- (void)setCurrentFilterIndex:(NSInteger)newIndex
+{
+    NSInteger index = [self currentFilterIndex];
+    if (newIndex == index) {
+        return;
+    }
+    [[NSUserDefaults standardUserDefaults] setObject:@(newIndex) forKey:CurrentPostListStatusFilterKey];
+    [NSUserDefaults resetStandardUserDefaults];
+
+    [self.recentlyTrashedPostIDs removeAllObjects];
+    [self updateFilterTitle];
+    [self updateAndPerformFetchRequestClearingCachedRowHeights:YES];
+}
+
+- (void)updateFilterTitle
+{
+    [self.filterButton setAttributedTitleForTitle:[self currentPostListFilter].title];
+}
+
+- (void)displayFilters
+{
+    NSMutableArray *titles = [NSMutableArray array];
+    for (PostListFilter *filter in self.postListFilters) {
+        [titles addObject:filter.title];
+    }
+    NSDictionary *dict = @{
+                          @"DefaultValue"   : [self.postListFilters firstObject],
+                          @"Title"          : NSLocalizedString(@"Filters", @"Title of the list of post status filters."),
+                          @"Titles"         : titles,
+                          @"Values"         : self.postListFilters,
+                          @"CurrentValue"   : [self currentPostListFilter]
+                          };
```

Comment:
Good point.  I like the initializer idea.  I think it would make the code a bit cleaner.